demonstrations/tutorial_qcbm.py

```diff
@@ -0,0 +1,327 @@
+#
+# .. math::
+#   C(\theta) = \sum_{\mathbf{x}} P_D(\mathbf{x}) \ln \left ( \frac{P_D(\mathbf{x})}{P_\theta(\mathbf{x})} \right)
+#
+#
+# Tensor Network Ansatz
+# ---------------------------
+# The algorithm presented in [#Benedetti]_, proposes the use of a hardware-efficient ansatz to prepare
+# the probability distribution using a quantum circuit. However, in this work we take inspiration from previous
+# approaches to generative modelling using tensor networks to represent the wave function, as done in [#Han]_
+# and [#Cheng]_ employing MPS and TTN, respectively. Since quantum circuits are a restricted class of tensor
+# networks, there is a natural relation between them that we can exploit to define a tensor-network inspired ansatz.
+# In particular, we take into consideration the local correlations of the data, and employ the MPS and
+# TTN circuit ansatz implemented in Pennylane. See this tutorial `this tutorial <https://pennylane.ai/qml/demos/tutorial_tn_circuits/>`_
+# for a deeper study of these ansätze. The conversion between the TTN class into a quantum circuit takes looks as
+# follows.
+#
+# .. figure:: ../_static/demonstration_assets/qcbm_tensor_network/ttn_ansatz.jpg
+#   :align: center
+#   :width: 70 %
+#
+# Analagously, converting an MPS tensor network into a quantum circuit would be done as shown in the following figure.
+#
+# .. figure:: ../_static/demonstration_assets/qcbm_tensor_network/mps_ansatz.jpg
+#   :align: center
+#   :width: 70 %
+#
+#
+# Pennylane implementation
+# ---------------------------
+# For this demo, we need a bit of an extra help from `another repo <https://github.com/XanaduAI/qml-benchmarks>`_ that
+# will allow us to generate the dataset needed in this tutorial:
+#
+# .. code-block:: bash
+#
+#    git clone https://github.com/XanaduAI/qml-benchmarks.git
+#    cd qml-benchmarks
+#    pip install .
+#
+# With that out of the way, we can start by importing what we need
+
+import pennylane as qml
+from pennylane import numpy as np
+from qml_benchmarks.data.bars_and_stripes import generate_bars_and_stripes
+import matplotlib.pyplot as plt
+
+##############################################################################
+# We then configure our random seed and setup some global variables that we will need for our experiments
+
+np.random.seed(1)
+dev = qml.device("default.qubit", wires=16)
+
+QUBITS = 16
+DATASET_SIZE = 1000
+TRAINING_ITERATIONS = 50
+
+MPS_DATA_SHAPE = (qml.MPS.get_n_blocks(range(QUBITS), 2), 6)
+TTN_DATA_SHAPE = (2 ** int(np.log2(QUBITS / 2)) * 2 - 1, 6)
+
+
+##############################################################################
+# For generating our dataset, which in our case is the Bar and Stripes dataset, we declare the following function:
+def prepare_dataset(size, rows, cols):
+    X, _ = generate_bars_and_stripes(size, rows, cols, noise_std=0.0)
+
+    # We need to transform the dataset from the {-1, 1} realm
```

Comment:
We got feedback from Juan Miguel about not adding comments inside the functions. They try to have the descriptions and comments outside